SecondStage/module28/src/test/java/page/googlecloud/GoogleCloudPlatformPricingCalculatorPage.java

```diff
@@ -0,0 +1,149 @@
+package page.googlecloud;
+
+import model.CloudPlatform;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import page.CommonAbstractPage;
+
+public class GoogleCloudPlatformPricingCalculatorPage extends CommonAbstractPage {
+
+    @FindBy (id = "myFrame")
+    private WebElement calculatorFrame;
+
+    @FindBy (xpath = "//md-tab-item/div[@title='Compute Engine']")
+    private WebElement computeEngineTab;
+
+    @FindBy (xpath = "//input[@ng-model='listingCtrl.computeServer.quantity']")
+    private WebElement numberOfInstancesTextBox;
+
+    @FindBy (xpath = "//md-select[@ng-model='listingCtrl.computeServer.os']")
+    private WebElement operatingSystemSoftwareDropdown;
+
+    @FindBy (xpath = "//md-select[@placeholder='VM Class']")
+    private WebElement machineClassDropdown;
+
+    @FindBy (xpath = "//md-select[@placeholder='Instance type']")
+    private WebElement machineTypeDropdown;
+
+    @FindBy (xpath = "//md-checkbox[@ng-model='listingCtrl.computeServer.addGPUs']")
+    private WebElement addGPUsCheckBox;
+
+    @FindBy (xpath = "//md-select[@placeholder='Number of GPUs']")
+    private WebElement numberOfGPUsDropdown;
+
+    @FindBy (xpath = "//md-select[@placeholder='GPU type']")
+    private WebElement typeOfGPUDropdown;
+
+    @FindBy (xpath = "//md-select[@placeholder='Local SSD']")
+    private WebElement localSSDDropdown;
+
+    @FindBy (xpath = "//md-select[@placeholder='Datacenter location']")
+    private WebElement datacenterLocationDropdown;
+
+    @FindBy (xpath = "//md-select[@placeholder='Committed usage']")
+    private WebElement committedUsageDropdown;
+
+    @FindBy (xpath = "//button[@aria-label='Add to Estimate' and @ng-disabled='ComputeEngineForm.$invalid || !listingCtrl.isGceAvailable']")
+    private WebElement addToEstimateButton;
+
+    public GoogleCloudPlatformPricingCalculatorPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public boolean isCalculatorPageFullyOpened() {
+        return new WebDriverWait(driver, 20)
```

Comment:
I suggest not use hardcoded wait intervals as it slows down your tests. Instead, you can use same interval defined on a base class and wait for some reference element to be visible/clickable, etc as an indicator of readiness